hooks/useThemeSettings/useThemeSettings.ts

```diff
@@ -0,0 +1,37 @@
+import { useNewsroomContext } from '@prezly/theme-kit-nextjs';
+import { useRouter } from 'next/router';
+import { useEffect } from 'react';
+import { useSessionStorage } from 'react-use';
+
+import type { ThemeSettings } from 'types';
+
+import { DEFAULT_THEME_SETTINGS, STORAGE_KEY } from './constants';
+import { parseQuery } from './utils';
+
+export function useThemeSettings(): ThemeSettings {
+    const { query } = useRouter();
+    const { themePreset } = useNewsroomContext();
+    const [previewSettings, setPreviewSettings] = useSessionStorage(STORAGE_KEY, {});
+
+    const settings: ThemeSettings = {
+        ...DEFAULT_THEME_SETTINGS,
+        ...themePreset?.settings,
+        ...previewSettings,
+    };
+
+    useEffect(() => {
+        const parsedQuery = parseQuery(query);
+        if (Object.keys(parsedQuery).length > 0) {
+            setPreviewSettings(parsedQuery);
+        }
```

Comment:
There's no extra logic to update already persisted settings, meaning that if you only provide one query parameter, we're gonna store only that one setting.

Since we always provide the full URL with all the parameters from our Prezly app, there's currently no need to support this.